pinot-plugins/pinot-input-format/pinot-avro-base/src/main/java/org/apache/pinot/plugin/inputformat/avro/AvroSchemaUtil.java

```diff
@@ -114,4 +137,39 @@ private static ArrayNode convertStringsToJsonArray(String... strings) {
     }
     return jsonArray;
   }
+
+  /**
+   * Applies the logical type conversion to the given Avro record field. If there isn't a logical
+   * type for the value then the value is returned unchanged. If there is a logical type associated
+   * to the field but no Avro conversion is known for the type then the value is returned unchanged.
+   *
+   * @param field Avro field spec
+   * @param value Value of the field
+   * @return Converted value as per the logical type in the spec, or the unchanged value if a
+   *     logical type or conversion can't be found.
+   */
+  public static Object applyLogicalType(Schema.Field field, Object value) {
+    if (field == null || field.schema() == null) {
+      return value;
+    }
+    LogicalType logicalType = LogicalTypes.fromSchemaIgnoreInvalid(field.schema());
+    if (logicalType == null) {
+      return value;
+    }
+    Conversion<?> conversion = AvroSchemaUtil.findConversionFor(logicalType.getName());
+    if (conversion == null) {
+      return value;
+    }
+    return Conversions.convertToLogicalType(value, field.schema(), logicalType, conversion);
+  }
+
+  static {
```

Comment:
(minor) Let's put this static block next to the map construction for readability